docs/docs/aws-s3-fileio-properties.md

```diff
@@ -0,0 +1,341 @@
+---
+title: "AWS S3 Properties"
+---
+<!--
+ - Licensed to the Apache Software Foundation (ASF) under one or more
+ - contributor license agreements.  See the NOTICE file distributed with
+ - this work for additional information regarding copyright ownership.
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+### Language Implementations
+
+| S3 Property                                                                               | Go | Java | Python | Rust |
+|-------------------------------------------------------------------------------------------|---|---|---|---|
+| [Progressive Multipart Upload](aws-s3-fileio-properties#progressive-multipart-upload)     |   | v |   |   |
```

Comment:
Pyiceberg fileIO implemenation replies on Pyarrow and fsspec (depends on user configuration to choose which one to use in the background in the end). AFAIK, both lib ([Pyarrow.fs.S3FileSystem](https://arrow.apache.org/docs/python/generated/pyarrow.fs.S3FileSystem.html) and [s3fs.core.S3FileSystem](https://s3fs.readthedocs.io/en/latest/api.html#s3fs.core.S3FileSystem) supports multipart upload.) 

I haven't checked the other properties in this list, so there might be other things already supported by Python but didn't marked correctly.

Reply:
@TiansuYu Thank you for your feedback.

You're right about the capability of Python libraries (`Pyarrow` and `fsspec`).

At the moment, I only list s3 properties used in the Iceberg implementations (and therefore configurable in Iceberg). I didn't cover all possible s3 properties from the libraries used by Java/Python/Go/Rust.

Take Rust as another example, we can set the [storage class](https://opendal.apache.org/docs/rust/opendal/services/struct.S3.html#method.default_storage_class) in Rust's OpenDAL library, however, Iceberg's Rust [config parsing function](https://github.com/apache/iceberg-rust/blob/main/crates/iceberg/src/io/storage_s3.rs#L63) doesn't recognize it yet.